libsplinter/src/service/network/interconnect/mod.rs

```diff
@@ -0,0 +1,722 @@
+            warn!("Service Interconnect is no longer running");
+        }
+
+        self.dispatch_shutdown.shutdown();
+    }
+}
+
+#[cfg(test)]
+pub mod tests {
+    use super::*;
+
+    use protobuf::Message;
+
+    use std::cell::RefCell;
+    use std::collections::VecDeque;
+
+    use crate::mesh::{Envelope, Mesh};
+    use crate::network::connection_manager::{
+        AuthorizationResult, Authorizer, AuthorizerError, ConnectionManager,
+    };
+    use crate::network::dispatch::{
+        ConnectionId, DispatchError, Handler, MessageContext, MessageSender,
+    };
+    use crate::protos::service;
+    use crate::service::network::ServiceConnectionManager;
+    use crate::transport::{inproc::InprocTransport, Connection, Transport};
+
+    // Verify that the ServiceInterconnect properly receives messages from services, passes them to
+    // the dispatcher, and sends messages from the handlers to other services.
+    //
+    // ServiceInterconnect will receive a message from services test-service and pass it to
+    // ComponentTestHandler. This handler will validate it came from test-service. The handler will
+    // then send a message to the ServiceInterconnect to send the message back to test-service.
+    // This valdiates that messages can be sent and recieved over the ServiceInterconnect.
+    //
+    // This tests also validates that ServiceInterconnect can retrieve the list of services from
+    // the ServiceConnectionManager using the ServiceConnectionManagerConnector.
+    //
+    // Finally, verify that ServiceInterconnect can be shutdown by calling shutdown_and_wait.
+    //
+    // 1. Starts up a ServiceConnectionManager and receives a incomming service connection.
+    //
+    // 2. The ServiceInterconnect is created with component disaptcher that contains a Handler for
+    //    SERVICE messages. This Handler will echo the service processor message content back to
+    //    the origin.
+    //
+    //    The main thread will then block on waiting for the remote thread to complete its
+    //    assertions.
+    //
+    // 3. The service running in another thread will send a ServiceProcessorMessage with the bytes
+    //    "test_retrieve" and will wait to recv the echoed bytes back from the main thread. This
+    //    would only happen if the ServiceInterconnect received the message and dispatched it to
+    //    the correct handler.  That Handler then must use the MessageSender pass the response to
+    //    the ServiceInterconnect.  The ServiceInterconnect will then send the message to the
+    //    Service.
+    //
+    //    Before the ServiceInterconnect can dispatch the message it has received, it has to find
+    //    the associated service_id for the connection id that was returned from Mesh. When it
+    //    receives the message from the service thread, it will not have this information, so it
+    //    must request the connection_id to service_id information from the
```

Comment:
Make consistent:
either remove the underscores and capitalize "id" ("service ID" and "connection ID")
or `s/connection id/connection_id/` on line 491

Reply:
This is a test comment and a technical detail, so I'm going to leave this as is.